File: walletSavior/setup_github_dirs.py
#!/usr/bin/env python3
import os
from pathlib import Path

# Create both directories
paths = [
    r'E:\pdf\capston01\walletSavior\.github\workflows',
    r'E:\pdf\capston01\walletSavior\.github\ISSUE_TEMPLATE'
]

print("Creating directories...\n")
for p in paths:
    Path(p).mkdir(parents=True, exist_ok=True)
    print(f"✓ Created: {p}")

print("\n" + "="*60)
print("VERIFICATION - Directory Contents")
print("="*60)

# Verify
github_path = Path(r'E:\pdf\capston01\walletSavior\.github')

print("\n=== .github directory contents ===")
try:
    items = sorted(github_path.iterdir())
    for item in items:
        print(f"  {'[DIR]':6} {item.name}")
except Exception as e:
    print(f"  Error: {e}")

print("\n=== workflows directory contents ===")
try:
    workflows = list(Path(paths[0]).iterdir())
    if workflows:
        for item in sorted(workflows):
            print(f"  {'[FILE]':6} {item.name}")
    else:
        print("  (empty directory)")
except Exception as e:
    print(f"  Error: {e}")

print("\n=== ISSUE_TEMPLATE directory contents ===")
try:
    templates = list(Path(paths[1]).iterdir())
    if templates:
        for item in sorted(templates):
            print(f"  {'[FILE]':6} {item.name}")
    else:
        print("  (empty directory)")
except Exception as e:
    print(f"  Error: {e}")

print("\n" + "="*60)
print("✓ All directories created and verified successfully!")
print("="*60)

File: walletSavior/setup_github_dirs.js
const fs = require('fs');
const path = require('path');

const basePath = 'E:\\pdf\\capston01\\walletSavior\\.github';
const dir1 = path.join(basePath, 'workflows');
const dir2 = path.join(basePath, 'ISSUE_TEMPLATE');

console.log('Creating directories...\n');

// Create directories
fs.mkdirSync(dir1, { recursive: true });
console.log(`✓ Created: ${dir1}`);

fs.mkdirSync(dir2, { recursive: true });
console.log(`✓ Created: ${dir2}`);

console.log('\n' + '='.repeat(60));
console.log('VERIFICATION - Directory Contents');
console.log('='.repeat(60));

// Verify .github directory
console.log('\n=== .github directory contents ===');
try {
  const items = fs.readdirSync(basePath).sort();
  items.forEach(item => {
    const fullPath = path.join(basePath, item);
    const isDir = fs.statSync(fullPath).isDirectory();
    console.log(`  ${isDir ? '[DIR] ' : '[FILE]'} ${item}`);
  });
} catch (e) {
  console.log(`  Error: ${e.message}`);
}

// Verify workflows directory
console.log('\n=== workflows directory contents ===');
try {
  const items = fs.readdirSync(dir1);
  if (items.length > 0) {
    items.sort().forEach(item => {
      console.log(`  [FILE] ${item}`);
    });
  } else {
    console.log('  (empty directory)');
  }
} catch (e) {
  console.log(`  Error: ${e.message}`);
}

// Verify ISSUE_TEMPLATE directory
console.log('\n=== ISSUE_TEMPLATE directory contents ===');
try {
  const items = fs.readdirSync(dir2);
  if (items.length > 0) {
    items.sort().forEach(item => {
      console.log(`  [FILE] ${item}`);
    });
  } else {
    console.log('  (empty directory)');
  }
} catch (e) {
  console.log(`  Error: ${e.message}`);
}

console.log('\n' + '='.repeat(60));
console.log('✓ All directories created and verified successfully!');
console.log('='.repeat(60));
